src/components/WorldMap.js

```diff
@@ -48,33 +58,50 @@ export default function WorldMap() {
       });
   }, []);
 
-  let textContent;
-  if (visitedCountries === 0) {
-    textContent = "You didn't select any Country yet";
-  } else if (visitedCountries === 1) {
-    textContent = `You have visited ${visitedCountries} Country in the World`;
-  } else {
-    textContent = `You have visited ${visitedCountries} Countries in the World`;
-  }
+  useEffect(() => {
+    const myMarkers = getMarkersFromLocalStorage();
+    setMarkers(myMarkers);
+    setNumberOfVisitedCountries(getCountriesCountFromLocalStorage());
+  }, []);
 
-  function handleClick(e, latlng) {
+  function handleClick(e, latlng, name) {
     if (e.target.checked === true) {
-      setVisitedCountries(visitedCountries + 1);
-      setMarkers([...markers, latlng]);
+      setNumberOfVisitedCountries(numberOfVisitedCountries + 1);
+      setMarkers([...markers, { latlng, name }]);
+      addMarkerToLocalStorage({ latlng, name });
+      addCountryCountToLocalStorage();
     } else {
-      setVisitedCountries(visitedCountries - 1);
+      setNumberOfVisitedCountries(numberOfVisitedCountries - 1);
       function filterFunction(country) {
-        return country !== latlng;
+        return country.name !== name;
       }
       const filteredMarkers = markers.filter(filterFunction);
       setMarkers(filteredMarkers);
+      removeMarkerFromLocalStorage(name);
+      removeCountryCountFromLocalStorage();
     }
   }
 
+  let textContent;
+  if (numberOfVisitedCountries === 0) {
+    textContent = "You didn't select any Country yet";
+  } else if (numberOfVisitedCountries === 1) {
+    textContent = `You have visited ${getCountriesCountFromLocalStorage()} Country in the World`;
+  } else {
+    textContent = `You have visited ${getCountriesCountFromLocalStorage()} Countries in the World`;
+  }
+
   function handleOnName(e) {
     setFilterInputValue(e.target.value);
   }
```

Comment:
That code snippet isn't used in the component.

Reply:
I have to call this function in order to filter the name of the country:
``
 <input
        placeholder="Type the name of the country"
        className="filterInput"
        value={filterInputValue}
        id="filterInput"
        onChange={handleOnName}
      ></input>
``